### ISSUE TITLE
'WebRTC/RTCCameraVideoCapturer.h' file not found

### ISSUE THREAD

**Issue by fabioplaka** [react-native-webrtc/react-native-webrtc#910]
Hello,

I have followed all the steps for installation of the library on iOS as it's described in your official documentation: https://github.com/react-native-webrtc/react-native-webrtc/blob/master/Documentation/iOSInstallation.md

When I run the application I get an error: **'WebRTC/RTCCameraVideoCapturer.h'** file not found

I use:  _react-native: 0.59.10_

Also i have added: **pod 'react-native-webrtc', :path => '../node_modules/react-native-webrtc'** on Podfile and then _pod install_

Cocoapods version is: **COCOAPODS: 1.5.3**

Please can you give me a solution for this error?

Thanks in advance

**Comment by brunsy** [comment]
There was an identical issue to this that was closed with a solution https://github.com/react-native-webrtc/react-native-webrtc/issues/909

**Comment by saghul** [comment]
As the documentation says, you need CocoaPods 1.10

**Comment by fabioplaka** [comment]
@saghul I just updated Cocoapods to 1.10 and the error still appear

![Screen Shot 2020-12-04 at 3 51 30 PM](https://user-images.githubusercontent.com/32648013/101178031-c5dbe580-3648-11eb-8593-1716b47ff49e.png)

**Comment by fabioplaka** [comment]
Perfect!
After installing  CocoaPods 1.10, and uninstalling and installing again pods, the error is gone

**Comment by benpeke** [comment]
@saghul I am still having this issue on our build fleet after updating to 1.10.x, could you give some more context on what is causing the problem so we can debug?

**Comment by saghul** [comment]
It was a problem in CocoaPods handling XCFramework.

**Comment by callmemonky** [comment]
@saghul I updated the cocoa pods to 1.10.x still its giving same error.
@benpeke got any solutions?

**Comment by GitPut** [comment]
Im still having the same issue for that past 2 days and cannot fix it. I have updated everything. Deleted node modules and installed them again. Deleted and install all pods again. I do not know what to do anymore

**Comment by benpeke** [comment]
Cocoapods will generate a Copy XCFramework script build phase, you can check this in the Pod xcodeproj in the react-native-webrtc target, and Cocoapods should handle linking the XCFramework. The issue I am having is with our build fleet which is never running that copy script, and then when it gets to the compilation of the react-native-webrtc code it fails. While I have not yet found a solution myself, I would suggest checking your Pod project and making sure it has that Build Phase.

**Comment by callmemonky** [comment]
@saghul it's still not fixed

**Comment by saghul** [comment]
@callmemonky I cannot reproduce the problem. There is a sample app in the project, can you reproduce it with it?

**Comment by dofl** [comment]
Having the same issues where this is used in Jitsi Meet: https://github.com/jitsi/jitsi-meet/issues/8517

**Comment by GitPut** [comment]
Good news I got it to kinda work in release mode. Afters days go going at this thing finally I found out that somehow it will not work on simulator which I have no fix for. But if you use a device and set it to release mode it works and is speedy fast!!

**Comment by benpeke** [comment]
I found the error I was having. The XCFramework is compressed to [WebRTC.xcframework.tgz](https://github.com/react-native-webrtc/react-native-webrtc/blob/master/apple/WebRTC.xcframework.tgz), for whatever reason this was not being automatically decompressed in our command line build which caused Cocoapods to ignore it completely and eventually get the missing header error. I just made a simple change to check if it was compressed and un-tar it after it was installed, after everything works perfectly.

**Comment by gowtham171** [comment]
cocoapod ver 1.10 still the issue is persist could any one help me to get rid of this PFA
![Screenshot 2021-05-18 at 7 29 56 PM](https://user-images.githubusercontent.com/26375829/118664882-76acfa00-b80f-11eb-9779-a4af6add0827.png)

**Comment by saghul** [comment]
Please read the comments above.

**Comment by JijoBose** [comment]
> I found the error I was having. The XCFramework is compressed to [WebRTC.xcframework.tgz](https://github.com/react-native-webrtc/react-native-webrtc/blob/master/apple/WebRTC.xcframework.tgz), for whatever reason this was not being automatically decompressed in our command line build which caused Cocoapods to ignore it completely and eventually get the missing header error. I just made a simple change to check if it was compressed and un-tar it after it was installed, after everything works perfectly.

@benpeke 
Could you please explain the changes (code) you added to fix this issue?

**Comment by benpeke** [comment]
> > I found the error I was having. The XCFramework is compressed to [WebRTC.xcframework.tgz](https://github.com/react-native-webrtc/react-native-webrtc/blob/master/apple/WebRTC.xcframework.tgz), for whatever reason this was not being automatically decompressed in our command line build which caused Cocoapods to ignore it completely and eventually get the missing header error. I just made a simple change to check if it was compressed and un-tar it after it was installed, after everything works perfectly.
> 
> @benpeke
> Could you please explain the changes (code) you added to fix this issue?

Ya i just wrote a little shell script to check if the file was compressed by checking the file extension. If it was compressed I just uncompressed with tar or gunzip

**Comment by bakdauletdinislam** [comment]
@benpeke Could you please explain the changes (code) you added to fix this issue?

**Comment by buddhiv** [comment]
@benpeke please add your code changes.

Aren't we having this again on cocoapods 1.10.x? This issue should be open again.

**Comment by jerryriff** [comment]
I'm also having the same issue.  Only in simulator, pod ver 1.10.1.

**Comment by caesar4321** [comment]
How's this going?

**Comment by irohitb** [comment]
I have m1 macbook and I fixed it by running it with Rosseta.

**Comment by edritech93** [comment]
same issue for me, I have m1 macbook
I try run with and without roseta, still problem

> I have m1 macbook and I fixed it by running it with Rosseta.

**Comment by arioki1** [comment]
I still get the same error.
Do you have a solution? @benpeke and @fabioplaka

**Comment by raymondpebble** [comment]
Seeing the same issue on m1 with the iOS simulator. 

Similar to @irohitb, I was able to build successfully by running xcodebuild in Rosetta. e.g. from terminal, `arch -x86_64 npm run ios`

Is there a way to get this running natively / could support be added?

**Comment by egulhan** [comment]
I'm also getting this error either when I try to run the app on an iOS simulator (with release build config) because I need to generate a simulator build for Facebook app review. is there no solution for that yet?

**Comment by diskshima** [comment]
For anyone , I had the same issue on iOS simulator running on M1.

TL;DR Upgrade to [1.94.0](https://github.com/react-native-webrtc/react-native-webrtc/releases/tag/1.94.0) or later (preferably the subminor updated 1.94.1).

Seems like the WebRTC.xcframework did not have bits necessary for an ARM64 simulator (the default one you get on M1).
saghul has done the job with this commit (ff9cc50b1b55263ef7f44dec2b6daf9ae2679623) so upgrading to 1.94.0 or later should fix the issue (at least did for me 😉 ).

**Comment by vkiranmaniya** [comment]
I managed to fix this issue by switching the architecture of `react-native-webrtc`. Here is how you can do that. I was having the same issue but only on simulator. This is how I fixed it.

![Screenshot 2022-03-03 at 5 02 50 PM](https://user-images.githubusercontent.com/22276302/156557104-72f63765-add0-41b2-ae26-391fd03c6f37.png)
